sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -74,9 +74,9 @@ private[sql] class HiveSessionCatalog(
   // | Methods and fields for interacting with HiveMetastoreCatalog |
   // ----------------------------------------------------------------
 
-  override def getDefaultDBPath(db: String): String = {
-    val defaultPath = context.hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE)
-    new Path(new Path(defaultPath), db + ".db").toString
+  override def getDatabasePath(dbName: String, path: Option[String] = None): String = {
+    val dbPath = path.getOrElse(context.hiveconf.getVar(HiveConf.ConfVars.METASTOREWAREHOUSE))
+    new Path(new Path(dbPath), dbName.toLowerCase() + ".db").toString
```

Comment:
The behavior of the current master when we dropping the database `default`. It is right. 
```
Error in query: org.apache.hadoop.hive.ql.metadata.HiveException: MetaException(message:Can not drop default database);
```
Will add a test case for this negative case.